src/csharp/ext/grpc_csharp_ext.c

```diff
@@ -1244,7 +1244,9 @@ GPR_EXPORT void* GPR_CALLTYPE grpcsharp_slice_buffer_adjust_tail_space(
   }
 
   if (buffer->count == 0) {
-    return NULL;
+    // when sending messages, C-core requires that there
+    // is at least on slice, even for empty payload.
+    grpc_slice_buffer_add_indexed(buffer, grpc_empty_slice());
```

Comment:
LGTM, but I do think it would be preferable to fix this in the core API. Can we file a bug to handle this in C-core (after which we should be able to remove this line) ?

Reply:
Filed https://github.com/grpc/grpc/issues/21299 and added a comment.